src/main/scala/cromwell/engine/EngineFunctions.scala

```diff
@@ -1,15 +1,24 @@
 package cromwell.engine
 
 import cromwell.binding.WdlFunctions
-import cromwell.binding.values.{WdlFile, WdlInteger, WdlString, WdlValue}
+import cromwell.binding.values._
 
-import scala.util.Try
+import scala.util.{Failure, Try}
 
 trait EngineFunctions extends WdlFunctions {
   protected def read_int(params: Seq[Try[WdlValue]]): Try[WdlInteger]
   protected def read_string(params: Seq[Try[WdlValue]]): Try[WdlString]
-  protected def stdout(params: Seq[Try[WdlValue]]): Try[WdlFile]
-  protected def stderr(params: Seq[Try[WdlValue]]): Try[WdlFile]
+  protected def stdout(params: Seq[Try[WdlValue]]): Try[WdlFileLike]
+  protected def stderr(params: Seq[Try[WdlValue]]): Try[WdlFileLike]
+
+  /**
+   * Extract a single `WdlValue` from the specified `Seq`, returning `Failure` if the parameters
+   * represent something other than a single `WdlValue`.
+   */
+  protected def extractSingleArgument(params: Seq[Try[WdlValue]]): Try[WdlValue] = {
```

Comment:
Would this be better as `extractFirstArgument` instead of `Single`? 

Also, can you explain the purpose behind this, i.e. why we want to grab only the head and why we don't care about the rest?